### PR TITLE
Improve performance of subscription panel

### PR DESCRIPTION
**Pull request by BentiGorlich**
- move the array sorting to SQL for performance reasons

### CODE REVIEW

File: src/Twig/Components/SidebarSubscriptionComponent.php [hunk 1]
```diff
@@ -25,22 +26,22 @@ class SidebarSubscriptionComponent
...
-        }
         if (ThemeSettingsController::ALPHABETICALLY === $this->sort) {
-            usort($this->magazines, fn ($a, $b) => $a->name > $b->name ? 1 : -1);
+            $this->magazines = $this->magazineRepository->findMagazineSubscriptionsOfUser($this->user, SubscriptionSort::Alphabetically, $max);
         } else {
-            usort($this->magazines, fn ($a, $b) => $a->lastActive < $b->lastActive ? 1 : -1);
+            $this->magazines = $this->magazineRepository->findMagazineSubscriptionsOfUser($this->user, SubscriptionSort::LastActive, $max);
         }
-        if (\sizeof($this->magazines) > $max) {
+        if (\sizeof($this->magazines) === $max) {
```
Comment (melroy89) [inline src/Twig/Components/SidebarSubscriptionComponent.php:43]: sorry equal to max?

Comment (melroy89) [reply src/Twig/Components/SidebarSubscriptionComponent.php:43]: maybe `>=` ?

Comment (melroy89) [reply src/Twig/Components/SidebarSubscriptionComponent.php:43]: is there a type check in PHP like `>==` xD?

File: src/Twig/Components/SidebarSubscriptionComponent.php [hunk 2]
```diff
@@ -25,22 +26,22 @@ class SidebarSubscriptionComponent
...
             $this->tooManyMagazines = true;
         }
-        $this->magazines = \array_slice($this->magazines, 0, $max);
```
Comment (melroy89) [inline src/Twig/Components/SidebarSubscriptionComponent.php:44]: yea this of course was indeed not ideal...  lol

File: src/Twig/Components/SidebarSubscriptionComponent.php (hunk 1 above)
Comment (BentiGorlich) [reply src/Twig/Components/SidebarSubscriptionComponent.php:43]: The reason is, the max is 50, so the array can never be bigger than 50. Before the change the array was not limited in size

File: src/Twig/Components/SidebarSubscriptionComponent.php (hunk 2 above)
Comment (BentiGorlich) [reply src/Twig/Components/SidebarSubscriptionComponent.php:44]: yep. Fetching **all** magazines, then sorting them and then discarding all but 50 of them is really not ideal performance wise

File: src/Twig/Components/SidebarSubscriptionComponent.php (hunk 1 above)
Comment (melroy89) [reply src/Twig/Components/SidebarSubscriptionComponent.php:43]: Yea ok..

File: src/Twig/Components/SidebarSubscriptionComponent.php (hunk 2 above)
Comment (melroy89) [reply src/Twig/Components/SidebarSubscriptionComponent.php:44]: Not exactly.. In fact first fetching all magazines by looping over all elements and adding them manually to an array is also less than ideal xD

**Pull request merged**